plugins/BulkTracking/Tracker/Handler.php

```diff
@@ -96,4 +104,23 @@ private function isTransactionSupported()
         return (bool) TrackerConfig::getConfigValue('bulk_requests_use_transaction');
     }
 
+    private function isBulkTrackingRequestAuthenticated(RequestSet $requestSet)
+    {
+        $tokenAuth = $requestSet->getTokenAuth();
+        if (empty($tokenAuth)) {
+            return false;
+        }
+
+        Piwik::postEvent('Request.initAuthenticationObject');
+
+        /** @var \Piwik\Auth $auth */
+        $auth = StaticContainer::get('Piwik\Auth');
+        $auth->setTokenAuth($tokenAuth);
+        $auth->setLogin(null);
+        $auth->setPassword(null);
+        $auth->setPasswordHash(null);
+        $access = $auth->authenticate();
+
+        return $access->getCode() != AuthResult::FAILURE;
```

Comment:
> Imagine someone is running a Piwik with many different, independent users (like a Piwik Cloud) in one big database... they could possibly see each other's invalid reports because each one has view access to that instance?

(thinking out loud) This case is hard for me to imagine... So a user uses bulk tracking w/ view access to a single piwik instance that has many users and many sites. The bulk tracking request includes requests for sites the user does not have access to.

One case is those individual requests have the correct token auths for the other users (because the user issuing the bulk tracking request is an attacker I guess). In this case the tracking succeeds for those requests. Any invalid requests will be identified (ie, non-existent site), but if the user can get other users token auths, they can get this information already?

Another case is those individual requests do not have token auths for other users. Authentication fails, and the attacker can get information about non-existent sites. This, I think, isn't a big deal by itself, the information is not terribly useful. The attacker could use id sites that he doesn't have access to in order to see what other id sites there are, but maybe this would already be possible through the UI/API? Not sure.

This assumes that the bulk requests requires authentication setting is off, which shouldn't normally be the case for such a setup, but it's not obvious that it should be set.

Well, I don't think it really matters right now whether this is only for superusers or not. W/ the log importer script, it should always be a superuser, unless the `--login --password` options are used w/ non-superuser. We can deal w/ that later if anyone actually requests it. **So I'll make it check for super users only.**